backend/endpoints/dependency_factory.go

```diff
@@ -25,14 +26,16 @@ type (
 
 	// DependencyProvider is a simple implementation of the dependency factory that supports the injection of "dynamic" dependencies
 	DependencyProvider struct {
-		Log        *logger.Log
-		FrontEndID int
+		Log         *logger.Log
+		FrontEndID  uuid.UUID
+		LogicalName string
+		URL         string
 	}
 )
 
 // GetFilesystemRepo is the constructor for FS repos
-func (dp DependencyProvider) GetFilesystemRepo() repos.FilesystemRepository {
-	return repos.NewFilesystemRepo(contexts.GetDatabaseContext())
+func (dp DependencyProvider) GetFilesystemRepo() (repos.FilesystemRepository, error) {
```

Comment:
IMO the dependency provider should be decorating the error returned by the repository factory